crates/compiler/src/visitors/constant_value_visitor.rs

```diff
@@ -134,7 +134,7 @@ impl From<Value> for ConstantValue {
 
 impl Default for ConstantValue {
     fn default() -> Self {
-        panic!("The `ConstantValueVisitor` was called in an unexpected context. This is a bug. Please report it at https://github.com/Mafii/rusty-yarn-spinner/issues/new")
+        unreachable!("The `ConstantValueVisitor` was called in an unexpected context. This is a bug. Please report it at https://github.com/Mafii/rusty-yarn-spinner/issues/new")
```

Comment:
Is `unreachable` correct here?
So we can never call `ConstantValue::default()`?

I assume this struct is not exposed to the user?

Reply:
The constant value visitor is meant to be used for very specific contexts, e.g. it is allowed to be called for a line but not for an entire dialog. This default implementation is called when the visitor is called in an unexpected way, which in the current implementation can indeed not happen. If we refactor the code wrongly, this panic will be reached and tell us. I'll write this into a comment in the code later.
With this explanation in mind, do you think I should change the unreachable into a panic?

Reply:
With a comment it should be fine, as we're not using `uncreachable_unchecked` here :)

Reply:
Copy-pasted my response into a comment 👍